.circleci/ci/src/jobs/backend/job-backend-build-and-publish-artifactory.ts

```diff
@@ -0,0 +1,43 @@
+import { commands, Config, Job, reusable } from '@circleci/circleci-config-sdk';
+import { OpenJdkExecutor } from '../../executors';
+import { Command } from '@circleci/circleci-config-sdk/dist/src/lib/Components/Commands/exports/Command';
+import { PrepareGpgCmd, RestoreMavenJobCacheCommand, SaveMavenJobCacheCommand } from '../../commands';
+import { config } from '../../config';
+import { CircleCIEnvironment } from '../../pipelines';
+
+export class BackendBuildAndPublishOnArtifactoryJob {
+  private static jobName = 'job-backend-build-and-publish-artifactory';
+
+  public static create(dynamicConfig: Config, environment: CircleCIEnvironment): Job {
+    const restoreMavenJobCacheCommand = RestoreMavenJobCacheCommand.get();
+    dynamicConfig.addReusableCommand(restoreMavenJobCacheCommand);
+
+    const prepareGpgCommand = PrepareGpgCmd.get(dynamicConfig);
+    dynamicConfig.addReusableCommand(prepareGpgCommand);
+
+    const saveMavenJobCacheCommand = SaveMavenJobCacheCommand.get();
+    dynamicConfig.addReusableCommand(saveMavenJobCacheCommand);
+
+    const steps: Command[] = [
+      new commands.Checkout(),
+      new reusable.ReusedCommand(restoreMavenJobCacheCommand, { jobName: 'release' }),
+      new commands.workspace.Attach({ at: '.' }),
+      new reusable.ReusedCommand(restoreMavenJobCacheCommand, { jobName: 'job-backend-build-and-publish-artifactory' }),
+      new commands.Run({
+        name: 'Remove `-SNAPSHOT` from versions',
+        command: `mvn -B versions:set -DremoveSnapshot=true -DgenerateBackupPoms=false
+sed -i "s#<changelist>.*</changelist>#<changelist></changelist>#" pom.xml`,
+      }),
+      new reusable.ReusedCommand(prepareGpgCommand),
+      new commands.Run({
+        name: "Maven deploy to Gravitee's private Artifactory",
+        command: `export BUILD_ID=${environment.buildId}
+export BUILD_NUMBER=${environment.buildNum}
+export GIT_COMMIT=$(git rev-parse --short HEAD)
+mvn --settings ${config.maven.settingsFile} -B -U -P all-modules,gio-artifactory-release,gio-release clean deploy -DskipTests=true -Dskip.validation -T 4 --no-transfer-progress`,
+      }),
```

Comment:
why not using `environment` parameter in the Run object ?

```suggestion
      new commands.Run({
        name: 'Maven deploy to Gravitee's private Artifactory',
        command: `mvn --settings ${config.maven.settingsFile} -B -U -P all-modules,gio-artifactory-release,gio-release clean deploy -DskipTests=true -Dskip.validation -T 4 --no-transfer-progress`,
        environment: {
          BUILD_ID: environment.buildId,
          BUILD_NUMBER: environment.buildNum,
          GIT_COMMIT: environment.sha1,
        },
      }),
```